File: doc-examples/mongo-example-kotlin/src/main/kotlin/example/CountryRegion.kt
/*
 * Copyright 2017-2020 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Relation
import io.micronaut.data.model.naming.NamingStrategies

// tag::namingStrategy[]
@MappedEntity(namingStrategy = NamingStrategies.Raw::class)
class CountryRegion(
        // end::namingStrategy[]
        @GeneratedValue
        @field:Id
        val id: Long?,
        val name: String,
        @field:Relation(Relation.Kind.MANY_TO_ONE)
        val country: Country)

File: doc-examples/jdbc-example-python/src/main/python/example/CountryRegion.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import GeneratedValue, Id, MappedEntity, Relation
from micronaut.data.model.naming import NamingStrategies

from example.Country import Country


# tag::namingStrategy[]
@MappedEntity(namingStrategy=NamingStrategies.Raw)
@dataclass
class CountryRegion:
    # end::namingStrategy[]
    name: str
    country: Annotated[Country | None, Relation("MANY_TO_ONE")] = None
    id: Annotated[int | None, Id, GeneratedValue] = None
    # tag::namingStrategy[]
    # ...
# end::namingStrategy[]
